Conditional DataStore Tab for Datapusher

JVickery-TBS opened the pull request:
feat(templates): conditional DataStore tab;

- Added helper for valid datapusher formats.
- Added condition to display the DataStore tab in Resource Edit.

### Features:

Conditions the DataStore tab in resource edit view for the Datapusher extension. Only show the tab if the resource is a supported datapusher format, or if the resource already has an existing datastore table (is datastore active).

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [X] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

JVickery-TBS commented:
Cypress failed in last workflow. Could someone re-run the workflow for me? Does not look like the Cypress failed because of code in PR.

wardi commented:
for xloader/datapusher we also need to watch out for datastore-first resources. Any time the `url_type` is set to one of these values we should hide the tab as well (and ideally refuse to run automatic loading for these resources).

There's a new `datastore_rw_resource_url_types` helper that returns a list of the datastore-first url_type values:
https://github.com/ckan/ckan/pull/7618/files

JVickery-TBS commented:
@wardi so I can just change condition to `( h.is_datapusher_format(res.format) or res.datastore_active ) and ( res.url_type not in h.datastore_rw_resource_url_types() )` in the template, and that should be good?

JVickery-TBS commented:
As suggested by @ThrawnCA I have added the DataStore action link along with the other action links in the `resource_read` template.

Added `datastore_rw_resource_url_types` helper to the conditions for the edit tab and the manage link button.

wardi commented on ckanext/datapusher/templates/package/resource_read.html:
```diff
@@ -0,0 +1,12 @@
+{% ckan_extends %}
+
+{% block action_manage %}
```
try inserting into the front of `action_manage_inner` so that it appears in the same order as the tabs when editing.

wardi replied:
add similar links in `package/snippets/resource_item.html` at the front of `resource_item_explore_inner` and `package/snippets/resources.html` at the front of `resources_list_edit_dropdown_inner` so that this tab appears along with the others for faster navigation.

These were recently added as part of #7590

JVickery-TBS replied:
@wardi cool! The conditional action link has been added to these template snippets in datapusher ext now. And `action_manage` updated to `action_manage_inner`

ThrawnCA commented on ckanext/datapusher/helpers.py:
```diff
@@ -29,3 +29,9 @@ def datapusher_status_description(status: dict[str, Any]):
         return _('Not Uploaded Yet')
+
+
+def is_datapusher_format(resource_format: str):
```
This helper could be expanded to cover the other two checks, `res.datastore_active` and `res.url_type not in h.datastore_rw_resource_url_types()`, since they're always used together.

JVickery-TBS replied:
Helpers and conditions combined into one helper.
